### ISSUE TITLE
Unable to build on macOS: "xcrun: error: unable to find utility xctest"

### ISSUE THREAD

**Issue by rsms** [val-lang/val#213]
What I did:

1. installed latest swift from https://www.swift.org/download/ (5.7.2)
2. `git clone https://github.com/val-lang/val.git`
3. `cd val`

```
$ swift build -c release
error: terminated(72): [...] PATH=/Users/rsms/bin:/Users/rsms/go/bin:/Users/rsms/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin [...] /usr/bin/xcrun --sdk macosx --find xctest output:
    xcrun: error: unable to find utility "xctest", not a developer tool or in PATH
$ uname -v
Darwin Kernel Version 19.6.0: Tue Jun 21 21:18:39 PDT 2022; root:xnu-6153.141.66~1/RELEASE_X86_64
$ swift --version
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin19.6.0
$ swiftc --version
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin19.6.0
```

(note: my entire `env` was printed, which I removed and replaced by `[...]` with the exception of `PATH`)

**Comment by dabrahams** [comment]
Not really a problem for Val to solve, but I think there are a number of answers [here](https://stackoverflow.com/questions/61501298/xcrun-error-unable-to-find-utility-xctest-not-a-developer-tool-or-in-path#comment124421829_61725799), especially probably the [last answer](https://stackoverflow.com/questions/61501298/xcrun-error-unable-to-find-utility-xctest-not-a-developer-tool-or-in-path#comment129581067_70829299).

**Comment by rsms** [comment]
Thank you for such a swift reply!

If `xctest` is required, then AFAIK Xcode is required to build Val on macOS, not just swift as the README suggests.

I'd recommend updating the installation section of the README to mention that Xcode is required.

https://github.com/val-lang/val/blob/934c869c73994e72007c6cee8ee6ca43b97ec5d7/README.md?plain=1#L12-L21

In total, this means I need to download ~13GB worth of tooling to build Val. Waiting for the download to complete and will let you know asap how it turned out.

**Comment by rsms** [comment]
Almost worked:

```shell
$ sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
Password:
$ swift build -c release
/Users/rsms/extsrc/val: error: package at '/Users/rsms/extsrc/val' is using Swift tools version 5.6.0 but the installed version is 5.3.0
$
```

This is weird because I installed swift 5.7.2 prior to installing xcode, but is seems that Xcode overwrote the swift installation:

```shell
$ swift --version
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin19.6.0
$
```

I'm using macOS 10.15 (can't upgrade) and so the most recent version of Xcode available for this system is 12.4.

Installing swift 5.7.2 again didn't work either for some reason. I've inspected what the installer contains and if I invoke that `swiftc --version` executable it does report the correct version (but not after installing.)

Turns out `/usr/bin/swift` is not the "real" path:

```shell
$ stat -f%z /usr/bin/swift
31488
```

Exploring further—and finally—after another web search, I learned that I have to find an identifier for the swift toolchain needed and then export an env var indicating that:

```shell
$ /usr/libexec/PlistBuddy -c "Print CFBundleIdentifier:" /Library/Developer/Toolchains/swift-latest.xctoolchain/Info.plist
org.swift.57202212131a
$ /usr/libexec/PlistBuddy -c "Print CFBundleIdentifier:" /Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain/Info.plist
org.swift.57202212131a
$ export TOOLCHAINS=org.swift.57202212131a
$ swift build -c release
2023-01-06 19:02:58.488 xcodebuild[91223:1300598] [MT] DVTToolchain: Failed to register toolchain: <DVTToolchain:0x7f87f350b440:'org.swift.57202212131a':'Swift 5.7 Development Snapshot 2022-12-13 (a)'>: Error Domain=DVTFoundationErrorDomain Code=-1 "Can't register <DVTFilePath:0x7f87f346c790:'/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>; identifier org.swift.57202212131a conflicts with <DVTFilePath:0x7f87f346d4f0:'/Users/rsms/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>" UserInfo={NSLocalizedDescription=Can't register <DVTFilePath:0x7f87f346c790:'/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>; identifier org.swift.57202212131a conflicts with <DVTFilePath:0x7f87f346d4f0:'/Users/rsms/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>}
2023-01-06 19:02:59.350 xcodebuild[91226:1300784] [MT] DVTToolchain: Failed to register toolchain: <DVTToolchain:0x7fdbc7c632c0:'org.swift.57202212131a':'Swift 5.7 Development Snapshot 2022-12-13 (a)'>: Error Domain=DVTFoundationErrorDomain Code=-1 "Can't register <DVTFilePath:0x7fdbc7c5d170:'/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>; identifier org.swift.57202212131a conflicts with <DVTFilePath:0x7fdbc7c5ded0:'/Users/rsms/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>" UserInfo={NSLocalizedDescription=Can't register <DVTFilePath:0x7fdbc7c5d170:'/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>; identifier org.swift.57202212131a conflicts with <DVTFilePath:0x7fdbc7c5ded0:'/Users/rsms/Library/Developer/Toolchains/swift-5.7.2-RELEASE.xctoolchain'>}
Fetching https://github.com/apple/swift-argument-parser.git
Fetching https://github.com/apple/swift-collections.git
# ...
Building for production...
[16/16] Linking valc
Build complete! (119.82s)
$ .build/x86_64-apple-macosx/release/valc --version
Error: Missing expected argument '<inputs> ...'
Help:  <inputs>
Usage: valc [--modules] [--import-builtin] [--no-std] [--typecheck] [--trace-inference <trace-inference>] [--emit <emit>] [-o <o>] [--verbose] [<inputs> ...]
  See 'valc --help' for more information.
$ 
```

It finally worked!

Wow, Swift is one messy toolchain
